src/rust/inetstack/protocols/tcp/active_open.rs

```diff
@@ -202,22 +207,18 @@ impl<N: NetworkRuntime> SharedActiveOpenSocket<N> {
             None => (0, 0),
         };
 
-        let rx_window_size: u32 = expect_ok!(
-            expect_some!(
-                (self.tcp_config.get_receive_window_size() as u32).checked_shl(local_window_scale as u32),
-                "TODO: Window size overflow"
-            )
-            .try_into(),
-            "TODO: Window size overflow"
+        // Expect is safe here because the receive window size is a 16-bit unsigned integer and MAX_WINDOW_SCALE is 14,
+        // so it is impossible to overflow the 32-bit unsigned int.
+        debug_assert!((local_window_scale as usize) < MAX_WINDOW_SCALE);
```

Comment:
I may have missed something here, but `local_window_scale` may be `MAX_WINDOW_SCALE`.

For instance, if `self.tcp_config.get_window_scale()` is greater than `MAX_WINDOW_SCALE`, `local` is truncated to `MAX_WINDOW_SCALE`,

If so, the debug assert should be `debug_assert!((local_window_scale as usize) <= MAX_WINDOW_SCALE)`